scripts/import.js

```diff
@@ -0,0 +1,67 @@
+const parse = require('csv-parse')
+const fs = require('fs')
+// const schemeSchema = require('../app/schema/scheme')
+const repo = require('../app/repository/scheme-repository')
+
+const processArgs = (args) => {
+  const config = {
+    filename: '',
+    noOfRecords: 3000
+  }
+  if (typeof args[2] === 'undefined') {
+    console.log('You must pass in a filename')
+    process.exit(1)
+  }
+  config.filename = args[2]
+  if (typeof args[3] !== 'undefined') {
+    const val = parseInt(args[3])
+    if (Number.isInteger(val)) {
+      config.noOfRecords = val
+    }
+  }
+  console.log(config)
+  return config
+}
+
+const config = processArgs(process.argv)
+let processed = 0
+
+const inStream = fs.createReadStream(config.filename)
+
+const parser = parse({
+  to: config.noOfRecords,
+  columns: true
+})
+
+parser.on('readable', async () => {
+  let record
+  while ((record = parser.read())) {
+    if (processed < config.noOfRecords) {
+      const importScheme = {
+        sbi: record.SBI,
+        agreementCode: record.AGREEMENT_CODE,
+        agreementDesc: record.AGREEMENT_DESC,
+        schemeOption: record.SCHEME_OPTION,
+        schemeOptionDesc: record.SCHEME_OPTION_DESC,
+        duration: record.DURATION,
+        agreementStartYear: record.AGREEMENT_START_YR,
+        agreementEndYear: record.AGREEMENT_END_YR,
+        agreementStartDate: record.AGREEMENT_START_DT,
+        agreementEndDate: record.AGREEMENT_END_DT,
+        parcelId: record.PARCEL_ID,
+        parcel: record.PARCEL,
+        hectares: record.HECTARES
+      }
+      // await schemeSchema.validateAsync(importScheme)
```

Comment:
Commented out code. I feel like we're not really the arbiters of what's valid in the data we're importing here so makes sense not to validate it at this point?

Reply:
Yes, the commented out code will be removed when this comes out of draft. This was a check to quickly highlight any issues with incoming data being invalid for coding reasons rather than poor data.